### DISCUSSION TITLE
Add Parameter for Security Scope Tags

### DISCUSSION THREAD

**Discussion by w0** [svrooij/WinTuner#106] [Ideas]
My org uses security scopes in Intune. The default scope is used by almost no-one outside of our full-intune admins. Please add a parameter to specify the security scope of an application being uploaded.

**Comment by svrooij** [comment]
Do you maybe have a link to the endpoint, or a dump from dev tools in the browser, where you see this properties being set?

I'm not an Intune admin and I don't know what you mean with the security scope.

**Reply by w0** [reply]
Is this helpful?

This is was the request:

https://graph.microsoft.com/beta/deviceManagement/roleScopeTags/getRoleScopeTagsById

`{"roleScopeTagIds":["5","9"]}`

This is the Response:

`{
    "@odata.context": "https://graph.microsoft.com/beta/$metadata#Collection(microsoft.graph.roleScopeTag)",
    "value": [
        {
            "id": "9",
            "displayName": "DEPT",
            "description": "Resources associated with the IT group \"DEPT\"",
            "isBuiltIn": false
        },
        {
            "id": "5",
            "displayName": "SHARED",
            "description": "Scope tag designating content meant to be available to all.",
            "isBuiltIn": false
        }
    ]
}`

**Reply by svrooij** [reply]
How does it add those to an app?
Is it a property when creating the app, is it set like an assignment?

Why I ask, the Intune APIs are mostly not documented and since I don't have these configured in my test tenant, I never saw a reference to them.

**Reply by w0** [reply]
Yes, when creating an app. It will ask if you want to add/remove any scope tags. It is page 7 for me. Right before group assignments.

**Reply by svrooij** [reply]
Do you have the body of the request where it adds the scope to the app, when you do it in the portal you can see all the requests in the dev tools

**Reply by w0** [reply]
I found this which would be helpful. This will get the scope available to a user so you can assign them as the default if none are specified. So if someone doesn't have access to the default scope they can upload apps.
/beta/deviceManagement/getEffectivePermissions(scope='*')

**Reply by w0** [reply]
And here is the full request body for a new app

```
{
    "@odata.type": "#microsoft.graph.win32LobApp",
    "applicableArchitectures": "x64",
    "allowAvailableUninstall": true,
    "categories": [],
    "description": "PuTTY release 0.81 (64-bit)",
    "developer": "",
    "displayName": "PuTTY release 0.81 (64-bit)",
    "displayVersion": "0.81.0.0",
    "fileName": "putty-64bit-0.81-installer.intunewin",
    "installCommandLine": "msiexec /i \"putty-64bit-0.81-installer.msi\" /qn",
    "installExperience": {
        "deviceRestartBehavior": "allow",
        "maxRunTimeInMinutes": 60,
        "runAsAccount": "system"
    },
    "informationUrl": "",
    "isFeatured": false,
    "roleScopeTagIds": [
        "5",
        "9"
    ],
    "notes": "",
    "minimumSupportedWindowsRelease": "1909",
    "msiInformation": {
        "packageType": "perMachine",
        "productCode": "{DDC95F26-92B1-4546-9678-5DC68DF76BA0}",
        "productVersion": "0.81.0.0",
        "requiresReboot": false,
        "upgradeCode": "{C9EAA861-2B72-4FAF-9FEE-EEB1AD5FD15E}",
        "productName": "PuTTY release 0.81 (64-bit)",
        "publisher": "Simon Tatham"
    },
    "owner": "",
    "privacyInformationUrl": "",
    "publisher": "Putty",
    "returnCodes": [
        {
            "returnCode": 0,
            "type": "success"
        },
        {
            "returnCode": 1707,
            "type": "success"
        },
        {
            "returnCode": 3010,
            "type": "softReboot"
        },
        {
            "returnCode": 1641,
            "type": "hardReboot"
        },
        {
            "returnCode": 1618,
            "type": "retry"
        }
    ],
    "rules": [
        {
            "@odata.type": "#microsoft.graph.win32LobAppProductCodeRule",
            "productVersionOperator": "notConfigured",
            "productCode": "{DDC95F26-92B1-4546-9678-5DC68DF76BA0}",
            "productVersion": null
        }
    ],
    "runAs32Bit": false,
    "setupFilePath": "putty-64bit-0.81-installer.msi",
    "uninstallCommandLine": "msiexec /x \"{DDC95F26-92B1-4546-9678-5DC68DF76BA0}\" /qn"
}
```

**Reply by svrooij** [reply]
You might be in for a treat with version `v1.0.1` 🎉

**Reply by w0** [reply]
I will give it a try. Thanks!

**Reply by w0** [reply]
Bumping into this error on 1.0.1, any suggestions? I ran

 New-WtWingetPackage -PackageId "Neovim.Neovim" -PackageFolder C:\packages | Deploy-WtWin32App -RoleScopeTags "MYTAG"

First run failure I get

```
INFO: [GraphAppUploader] Creating new Win32LobApp
Deploy-WtWin32App: The request is invalid.
Deploy-WtWin32App: The request is invalid.
```

Subsequent runs I got this 

```
INFO: [GraphAppUploader] Creating new Win32LobApp
Deploy-WtWin32App: Cannot create an abstract class.
Deploy-WtWin32App: Cannot create an abstract class.
```
